### DISCUSSION TITLE
Is sequence ordering guaranteed in connect-web in the same way as in protocol buffers?

### DISCUSSION THREAD

**Discussion by hiroyuki-fujii-td** [connectrpc/connect-es#710] [Q&A] [answered]
Thank you for your contribution!

I create proto file like below.
```
message GetRequest {
  repeated string name = 1;
}
```

Then I use connect-web, no use proxy like envoy.
So I create a connectClient.ts like below, along [this document](https://connect.build/docs/web/using-clients/).

```
import { createPromiseClient, type PromiseClient } from '@bufbuild/connect';
import { createConnectTransport } from '@bufbuild/connect-web';
import { type ServiceType } from '@bufbuild/protobuf';
import { useMemo } from 'react';

const transport = createConnectTransport({
  baseUrl: 'http://localhost:9991',
});

export function useClient<T extends ServiceType>(service: T): PromiseClient<T> {
  return useMemo(() => createPromiseClient(service, transport), [service]);
}
```
Client can request to connect server by using curl command.

I found this disucussion [Does repeated fields in protobuffers keep the order they are inserted?](https://stackoverflow.com/questions/55063424/does-repeated-fields-in-protobuffers-keep-the-order-they-are-inserted), recognized that order is guaranteed in gRPC using `connect.WithGRPC()`.

Is ordering guaranteed in the same way as gRPC using `connect.WithGRPC()`?
Is connect-web also compliant with [this Documentation](https://protobuf.dev/programming-guides/encoding/#optional) ?

best regard

**Comment by smaye81** [comment, marked as answer]
Hi @hiroyuki-fujii-td. Yes, the ordering for repeated fields is guaranteed with Connect-Web also according to the spec. The same guarantees apply, i.e:

> The order of the elements with respect to each other is preserved when parsing, though the ordering with respect to other fields is lost.
